pkg/scene/widget.go

```diff
@@ -21,17 +29,22 @@ func NewWidget(wid *id.WidgetID, plugin id.PluginID, extension id.PluginExtensio
 		wid = id.NewWidgetID().Ref()
 	}
 
+	if widgetLayout == nil {
+		widgetLayout = &WidgetLayout{}
+	}
+
 	return &Widget{
-		id:        *wid,
-		plugin:    plugin,
-		extension: extension,
-		property:  property,
-		enabled:   enabled,
+		id:           *wid,
+		plugin:       plugin,
+		extension:    extension,
+		property:     property,
+		enabled:      enabled,
+		widgetLayout: widgetLayout,
 	}, nil
 }
 
-func MustNewWidget(wid *id.WidgetID, plugin id.PluginID, extension id.PluginExtensionID, property id.PropertyID, enabled bool) *Widget {
-	w, err := NewWidget(wid, plugin, extension, property, enabled)
+func MustNewWidget(wid *id.WidgetID, plugin id.PluginID, extension id.PluginExtensionID, property id.PropertyID, enabled bool, widgetLayout *WidgetLayout) *Widget {
+	w, err := NewWidget(wid, plugin, extension, property, enabled, widgetLayout)
```

Comment:
This function has too many parameters, so consider turning it into a builder pattern.

Reply:
I am thinking of refactoring the widget, widget system and widget align system into a widget package and creating a builder at that point. If that sounds like a good idea I will leave this as is for now.